packages/peer-exchange/src/waku_peer_exchange_discovery.ts

```diff
@@ -149,60 +149,53 @@ export class PeerExchangeDiscovery
     }, queryInterval * currentAttempt);
   };
 
-  private query(peerId: PeerId): Promise<void> {
-    return this.peerExchange.query(
-      {
-        numPeers: DEFAULT_PEER_EXCHANGE_REQUEST_NODES,
+  private async query(peerId: PeerId): Promise<void> {
+    const peerInfos = await this.peerExchange.query({
+      numPeers: DEFAULT_PEER_EXCHANGE_REQUEST_NODES,
+      peerId,
+    });
+
+    for (const _peerInfo of peerInfos) {
+      const { ENR } = _peerInfo;
+      if (!ENR) {
+        log("no ENR");
+        continue;
+      }
+
+      const { peerId } = ENR;
+      const multiaddrs = ENR.getFullMultiaddrs();
+
+      if (!peerId || !multiaddrs || multiaddrs.length === 0) continue;
+
+      if (await this.components.peerStore.has(peerId)) continue;
```

Comment:
I think we can combine `has` and `getTags` here and use only `getTags` and handle an error. That way we save time not waiting for one async operation.